homeassistant/components/formula_one/sensor.py

```diff
@@ -0,0 +1,419 @@
+"""Support for Formula 1 sensors."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import ergast_py as ergast
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, F1_DISCOVERY_NEW, F1_STATE_MULTIPLE
+from .coordinator import F1UpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Configure sensors from a config entry created in the integrations UI."""
+
+    @callback
+    def async_setup_sensors() -> None:
+        _LOGGER.debug("Handling new drivers/constructors/races")
+
+        coordinator: F1UpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+        entities: list[F1Sensor] = []
+
+        for standing in coordinator.constructor_standings:
+            new_cp_sensor = F1ConstructorPositionSensor(coordinator, standing.position)
+            if new_cp_sensor.unique_id not in coordinator.entity_ids:
+                entities.append(new_cp_sensor)
```

Comment:
Don't create objects which are then just discarded. Instead, add a static method `F1ConstructorPositionSensor.calculate_entity_id` which is called to check if the sensor already exists.